regression/goto-gcc/dev_null1/main.c

```diff
@@ -0,0 +1,9 @@
+void gcc_fails(void)
+{
+  asm volatile ( ".if ((1 > 0) < 0); .error \"this is gcc\";.endif" );
```

Comment:
What is this peculiar magic test?

Reply:
It's modeled after code that Xen uses to figure out properties of the underlying assembler...

Reply:
Would 
```
#if defined(__GNUC__) && !defined(__clang__)
#error "GCC fails"
#endif
```
achieve the same effect?

Reply:
I don't think so, because it would (correctly) fail even without this patch.

Reply:
OK, why do you need a separate thing that makes GCC and Clang fail? Would a single ASM block that always `.error`s do?

I'm guessing the idea is we're looking for something that trips the native compiler but not goto-cc, which I guess ignores the odd ASM block?

Reply:
True, a single block would do. Will simplify.

Reply:
Done.